addons/godot_rl_agents/sensors/sensors_3d/GridSensor3D.gd

```diff
@@ -38,29 +38,55 @@ class_name GridSensor3D
 		cell_height = value
 		_update()	
 
-@export_range(1, 21, 2, "or_greater") var grid_size := 3:
-	get: return grid_size
+@export_range(1, 21, 2, "or_greater") var grid_size_x := 3:
+	get: return grid_size_x
 	set(value):
-		grid_size = value
+		grid_size_x = value
 		_update()
 
-var _obs_buffer : PackedByteArray
-var _box_shape : BoxShape3D
-var _collision_mapping : Dictionary
-var _n_layers_per_cell : int
+@export_range(1, 21, 2, "or_greater") var grid_size_z := 3:
+	get: return grid_size_z
+	set(value):
+		grid_size_z = value
+		_update()
+
+var _obs_buffer: PackedFloat64Array
+var _box_shape: BoxShape3D
+var _collision_mapping: Dictionary
+var _n_layers_per_cell: int
 
+var _highlighted_box_material: StandardMaterial3D
+var _standard_box_material: StandardMaterial3D
 
 func get_observation():
 	return _obs_buffer
-
-
+	
 func _update():
 	if Engine.is_editor_hint():
-		_spawn_nodes()	
-
+		if is_node_ready():
+			_spawn_nodes()	
 
 func _ready() -> void:
-	_spawn_nodes()
+	if Engine.is_editor_hint():	
```

Comment:
Before this change, trying to load (in Godot editor) a training scene containing 32 instances of a scene that uses the grid sensor with:

```
grid size x = 17
grid size z = 17
```

causes the Godot editor to become unresponsive, and after a while it will crash on my PC.

If `@tool` is commented out before loading the scene, it will load it quickly without issues.

The cause may be that it's trying to rebuild a large number of nodes for the grid sensor for each instance of the training scene.

This is an attempt to fix that. Nodes are still rebuilt when starting the game which may not be necessary in all cases, but it's possible that signals may still need to be connected, so I left that part unmodified for now.